src/service/mod.rs

```diff
@@ -0,0 +1,160 @@
+//! Implementations of messages sent to and from services
+
+use ast::ValueSequence;
+
+trait ServiceMessage where Self: Sized {
+    fn unmarshall(body: &ValueSequence) -> Result<Self, UnmarshallError>;
+    fn marshall(&self) -> ValueSequence;
+}
+
+#[derive(Debug, PartialEq)]
+enum UnmarshallError {
+    RequiredValueNotProvided,
+}
+
+mod whois {
```

Comment:
Consider breaking `whois` and `iam` into two files, unless they will always stay this small.